src/mixins/nft.js

```diff
@@ -423,6 +424,46 @@ export default {
           message: this.normalizeNFTMessage(m),
         }));
     },
+    recommendedList() {
+      const featuredSet = this.getNFTClassFeaturedSetByAddress(this.iscnOwner);
+      const hiddenSet = this.getNFTClassHiddenSetByAddress(this.iscnOwner);
+
+      let recommendedList =
+        this.getNFTListMapByAddress(this.iscnOwner)?.created || [];
+      const userCollected =
+        this.getNFTListMapByAddress(this.getAddress)?.collected || [];
+
+      if (hiddenSet) {
+        recommendedList = recommendedList.filter(
+          item =>
+            !userCollected.some(
+              collectedItem => collectedItem.classId === item.classId
+            ) &&
+            !hiddenSet.has(item.classId) &&
+            item.classId !== this.classId
+        );
+      } else {
+        recommendedList = recommendedList.filter(
+          item =>
+            !userCollected.some(
+              collectedItem => collectedItem.classId === item.classId
+            )
+        );
+      }
+
```

Comment:
```suggestion
      recommendedList = recommendedList.filter(
        item =>
          !userCollected.some(
            collectedItem => collectedItem.classId === item.classId
          ) && item.classId !== this.classId
      );
      if (hiddenSet) {
        recommendedList = recommendedList.filter(
          item =>
            !hiddenSet.has(item.classId)
        );
      }
```
Since the list is short, just do a 2 pass for better code structure?